doc/proposals/scorecard-custom-tests-2.md

```diff
@@ -0,0 +1,224 @@
+
+
+## Release Signoff Checklist
+
+- \[ \] Enhancement is `implementable`
+- \[ \] Design details are appropriately documented from clear requirements
+- \[ \] Test plan is defined
+- \[ \] Graduation criteria for dev preview, tech preview, GA
+- \[ \] User-facing documentation is created 
+
+## Summary
+
+This enhancement proposal outlines the SDK scorecard’s design for running of custom tests, specifically tests that are outside of what Red Hat includes by default within the scorecard’s tests.   The proposed design would provide both a means to write custom tests and also convert the existing scorecard tests (basic and olm) into the custom test format.
+
+## Motivation
+
+The scorecard needs a robust means of enabling end-users of the SDK to write and execute their own operator tests.   Ideally, the custom test framework would decouple test logic from within the SDK itself, making the SDK and scorecard logic in particular more flexible and maintainable in the long run.
+
+## Goals
+
+ * Document a design for scorecard to execute end-user or custom tests.
+
+ * Document stories/steps required to deliver scorecard custom test functionality.
+
+### Non-Goals
+
+ * This is a design proposal and does not include a proof-of-concept.
+
+## Proposal
+
+This design proposal is to establish a pattern for implementing custom scorecard tests and also support the goal of migrating existing scorecard tests into the same format as the custom tests.
+
+With this proposal, the scorecard will continue to setup and teardown the Operator being tested, as well as creating Custom Resources being tested.
+
+### User Stories
+
+#### Story 1
+
+Scorecard users will be able to construct custom tests and execute them using the scorecard.
+
+#### Story 2
+
+Scorecard will ship with an example custom test that end users could use as a reference when writing their custom tests.
+
+#### Story 3
+
+Scorecard internal tests will migrate to the custom test format and therefore be externalized from the scorecard binary itself.
+
+#### Story 4 (optional)
+
+Scorecard would support the creation of a Kuttl test image.  This custom test would execute kuttl to perform testing enabling scorecard users a means to write tests using kuttl but have the result integrated into scorecard results.  
+
+## Design Details
+
+### Custom Test Definition
+
+* A custom test is any sort of test logic required, written in any programming language that satifies the scorecard test requirements for output format and execution.
+
+### Custom Test Components
+
+This proposal outlines the end-user facing components that make up a custom test including:
+
+#### Test Image
+
+A custom test is based on a container image, that image is created by
+the end-user using whatever tools they want to use.
+
+The test image must produce log output that conforms to the scorecard Test Result output.
+
+The test image will be executed as a Pod by the scorecard with a restart policy of `never`.
```

Comment:
Did you consider `Job` instead of `Pod`. What are the pros/cons of each?

Reply:
I did, the only value that I could see from doing this with a Job would be for restarting a test, I didn't see a use case for job restarts with scorecard at this point.  I guess the con's of a job is that its another layer of abstraction to go find the pod log output as well for scorecard and end users alike.   If you can think of a good pro, I'd be for doing this as a Job, but I couldn't come up with one.

Reply:
That all sounds like good reasoning to use a Pod directly. :+1: